File: deepspark/spark16/src/main/scala/sparklearnscala/sql/source/ParquetOpts.scala
package sparklearnscala.sql.source

import org.apache.spark.api.java.JavaSparkContext
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.sql.SQLContext

object ParquetOpts {
  def main(args: Array[String]): Unit = {
    parquetRead;
  }
  
  def parquetRead():Unit = {
		val conf = new SparkConf().setAppName(ParquetOpts.getClass.getSimpleName()).setMaster("local");
		val sc = new SparkContext(conf);
		
		val sqlContext = new SQLContext(sc);

		//DataFrame df = sqlContext.read().parquet("data/user.parquet");
		//DataFrame df = sqlContext.read().format("parquet").load("data/user.parquet");
		
		val df = sqlContext.sql("SELECT * FROM parquet.`data/user.parquet`");

		df.show();
		//df.write().save("data/userparquet");
	}
}

File: deepspark/spark16/src/main/java/sparklearn/sql/source/ParquetOpts.java
/** 
 * Project Name:sparklearn 
 * File Name:ParquetOpts.java 
 * Package Name:sparklearn.sql 
 * Date:2017年7月12日下午4:27:50 
 * sky.zyl@hotmail.com
*/  
  
package sparklearn.sql.source;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;

/** 
 * ClassName:ParquetOpts <br/> 
 * Function: TODO ADD FUNCTION. <br/> 
 * Reason:   TODO ADD REASON. <br/> 
 * Date:     2017年7月12日 下午4:27:50 <br/> 
 * @author   hadoop 
 * @version   
 * @see       
 */
public class ParquetOpts {

	public static void main(String[] args) {
		parquetRead();

	}
	
	public static void parquetRead() {
		SparkConf conf = new SparkConf().setAppName(JsonOpts.class.getSimpleName()).setMaster("local");
		JavaSparkContext sc = new JavaSparkContext(conf);
		SQLContext sqlContext = new org.apache.spark.sql.SQLContext(sc);

		//DataFrame df = sqlContext.read().parquet("data/user.parquet");
		//DataFrame df = sqlContext.read().format("parquet").load("data/user.parquet");
		
		DataFrame df = sqlContext.sql("SELECT * FROM parquet.`data/user.parquet`");

		df.show();
		//df.write().save("data/userparquet");
	}

}
 